src/vtcomposite.cpp

```diff
@@ -602,75 +615,86 @@ struct LocalizeWorker : Napi::AsyncWorker
           baton_data_{std::move(baton_data)},
           output_buffer_{std::make_unique<std::string>()} {}
 
-    // for a given feature, determine how many clones of the feature are
-    // required given a worldview.
-    // - If the feature has _mbx_worldview: all, return {"all"}
-    // - If there is no requested worldview "null", determine which legacy worldviews should be created
-    std::vector<std::string> worldviews_for_feature(std::string const& pval) const
-    {
-        if (pval == "all")
-        {
-            return {"all"};
-        }
-
-        std::vector<std::string> matching_worldviews;
-        // convert pval into vector of strings US,CN => {"US", "CN"}
-        std::vector<std::string> worldview_values = utils::split(pval);
-        // worldview: XX
-        if (!baton_data_->worldviews.empty())
-        {
-            utils::intersection(worldview_values, baton_data_->worldviews, matching_worldviews);
-        }
-
-        return matching_worldviews;
-    }
-
-    // create a feature from a list of properties
-    // optionally define a worldview property if provided
-    static void build_localized_feature(
-        vtzero::feature const& feature,
+    // create a feature with new properties from a template feature
+    static void build_new_feature(
+        vtzero::feature const& template_feature,
         std::vector<std::pair<std::string, vtzero::property_value>> const& properties,
-        std::string const& worldview,
+        std::string const& worldview_key,
+        std::string const& worldview_val,
         vtzero::layer_builder& lbuilder)
     {
         vtzero::geometry_feature_builder fbuilder{lbuilder};
-        fbuilder.copy_id(feature); // todo deduplicate this (vector tile spec says SHOULD be unique)
-        fbuilder.set_geometry(feature.geometry());
+        fbuilder.copy_id(template_feature); // TODO: deduplicate this (vector tile spec says SHOULD be unique)
+        fbuilder.set_geometry(template_feature.geometry());
 
-        if (!worldview.empty())
-        {
-            fbuilder.add_property("worldview", worldview);
-        }
+        // add property to feature
         for (auto const& property : properties)
         {
-            // no-op on any property called "worldview" if worldviews have been
-            // created dynamically from _mbx_wordlview
-            if (!worldview.empty() && property.first == "worldview")
+            if (property.first == worldview_key) // safeguard – should always evaluate to false
```

Comment:
note to self: why?

Reply:
Because in the current implementation, when we encounter `{worldview_property}` or `{worldview_prefix}{worldview_property}` while looping through the properties, we only store the worldviews to create in the `worldviews_to_create` variable but not add them to the `final_properties` vector. After we're done looping through the properties, we decide how many clones of the feature to build (for now we are limiting to 1) and what each clone should have as its `{worldview_property}` . 

This safeguard is to prevent future code change that might accidentally/unintentionally add worldview to `final_properties`, and also to make clear for humans that `worldview_key` of the property always come from `worldview_val`.